### ISSUE TITLE
Global queryables conflict with STAC

### ISSUE THREAD

**Issue by m-mohr** [opengeospatial/ogcapi-features#830]
STAC has global Item search at /search. The Filter/CQL support for global Item search requires a /queryables endpoint.

Unfortunately, the link conflicts with https://github.com/opengeospatial/ogcapi-features/blob/master/proposals/search/global-list-of-queryables.adoc which also exposes a /queryables endpoint for /collections. 

If I try to implement queryables for /search and /collections via links, I can't distinguish which endpoint applies to /search and which applies to /collections.

How can we solve this conflict?

cc @philvarner

**Comment by ycespb** [comment]
Can't /collections/queryables be used for the list of collection search queryables ?  This does not create a conflict with /queryables, nor with /collections/{id}/queryables.  Our implementation at emc.spacebel.be is using this URL which can be found via the rel=".../queryables" as well in the /collections response.

**Comment by m-mohr** [comment]
It may conflict with a potential collection route, so I'd not recommend that by default, but individual implementations could go for that if we leave the path open. A better recommendation might be `/queryables/collections`.

**Comment by m-mohr** [comment]
Thanks, that works for me!

**Comment by pvretano** [comment]
This discussion sounds like the [Local Resources Catalogue](https://docs.ogc.org/DRAFTS/20-004.html#clause-local-resources-catalogue) conformance class is OGC API Records.  

That conformance class specifies how to extend existing endpoints like `/collections` or `/processes` (for example) to make them searchable.  So you can say something like `GET /collections?q=seabed&bbox=10,10,20,20` and the response would be the normal `/collections` response except that the only collections included in the response would be those that intersect the specified `bbox` and which contain the search term `seabed` is their titles or descriptions.

[OGC API Records ](https://docs.ogc.org/DRAFTS/20-004.html#_operation_4) indicates that the queryables endpoint  for local resources endpoints (like `/collections` or `/processes`) should be specified via a link (`rel=ogc::queryables`) in the links section.

As an aside (**and this is not part of any specification, its just something I'm experimenting with**), my implementation of OGC APIs provides a top-level `/queryables` endpoint with the following structure:
````
/queryables
|
+-- /collections
|   |
|   +-- {collectionId}
|       |
|       +-- /items
|       |
|       +-- /styles
|
+-- /processes
|
+-- /jobs
````

At any endpoint in this tree you get back a JSON-schema that lists the queryables for searching the corresponding endpoint in the OGC API resource tree.

| Queyrables endpoint                               |Target OGC API endpoint        | Description                                                    |             
|---------------------------------------------------|-----------------------------------|----------------------------------------------------------------|
| GET /queryables/collections                       | /collections                      | List of queryables for searching for collections               |
| GET /queryables/collections/{collectionId}/items  | /collections/{collectionId}/items | List of queryables for searching for items of this collection  |             
| GET /queryables/collections/{collectionId}/styles | /collections/{collectionId}/styles | List of queryables for searching for styles of this collection |
| GET /queryables/processes                         | /collections/processes            | List of queryables for searching for processes                 |             
| GET /queryables/jobs                              | /jobs                             | List of queryables for searching for jobs                      |

An empty response means that you can't search the corresponding endpoint in the OGC API resource tree.

The link to queryables in the various links section simply point to these endpoints.  So, the link in the `/collections` object that point to the queryables simply points to `/querables/collection`.

Anyway, just something I've been playing with in relations to queryables.

**Comment by cportele** [comment]
I like the idea of making all queryables resources (also) available under `/queryables`. Would we then add those that are valid for all collections in `/search` at `/queryables/search` or still at `/queryables` as previously planned? I think `/queryables/search` would be the better fit.

**Comment by m-mohr** [comment]
Makes sense to me and avoids potential conflicts. Is this meant to be a guideline/recommdation or a requirement? I would prefer the former, I think. 

In STAC we currently use /collections/:id/queryables, which I think originates from OGC. We can change this in our API extension, I think.

This structure would also be applied to something like /sortables, right?

**Comment by pvretano** [comment]
@m-mohr yes, `/sortables/collections/{collectionId}`, etc...

**Comment by cportele** [comment]
>  In STAC we currently use `/collections/:id/queryables`, which originates from one of the OGC APIs, I think. We can change this in our API extension though.

Features Part 3 will use that path (`/collections/{collectionId}/queryables`). 

The approach with `/queryables/...` and `/sortables/...` would be specified in the Schema extension. 

The same resource would then be available at `/collections/{collectionId}/queryables` (Part 3) and `/queryables/collections/{collectionId}/items` (Part n, Schemas).

**Comment by cportele** [comment]
Meeting 2024-08-26: The SWG members will review this again and discuss in the next meeting(s). There are also the queryables for the local resources catalog that we need to consider in this context.

**Comment by cportele** [comment]
Meeting 2025-03-24: @cportele will review, if this is relevant for Part 5 or if we can discuss this further in the context of Part 10.
